src/vs/workbench/browser/actions/inputResultsCommands.ts

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { KeyMod, KeyCode } from 'vs/base/common/keyCodes';
+import { ContextKeyExpr } from 'vs/platform/contextkey/common/contextkey';
+import { IInputResultsService, InputResultsViewInputFocusedContextKey, InputResultsViewResultsFocusedContextKey } from 'vs/platform/inputResults/browser/inputResultsService';
+import { KeybindingWeight, KeybindingsRegistry } from 'vs/platform/keybinding/common/keybindingsRegistry';
+import { RawWorkbenchListScrollAtTopContextKey, WorkbenchListFocusContextKey } from 'vs/platform/list/browser/listService';
+
+KeybindingsRegistry.registerCommandAndKeybindingRule({
+	id: 'inputResults.focusInput',
+	weight: KeybindingWeight.WorkbenchContrib,
+	when: ContextKeyExpr.and(
+		InputResultsViewResultsFocusedContextKey,
+		ContextKeyExpr.or(
+			WorkbenchListFocusContextKey?.negate(),
```

Comment:
I don't understand this line?

Reply:
Well, this is to achieve the following behavior:

Ctrl+UpArrow should focus "input box" if and only if 
1. The "results pane" has focus (`resultsFocused`), _and_
2. either of these conditions holds
    2.1 the focused widget on the "result pane" is not a list (`!listFocused`), _or_
    2.2 the focused widget on the "result pane" is a list, but its vertical scrollbar is at the top-most position (`listFocused && listScrollAtTop`).

So overall, the expression should be

`resultsFocused && ((!listFocused) || (listFocused && listScrollAtTop))`

which can be simplified into

`resultsFocused && ((!listFocused) || listScrollAtTop)`

Reply:
Is there a case today where the results view is not a list?

Reply:
Yes. For example in MarkersView, if there's no problems currently, the only widget on the results panel will be a messagebox displaying something like "No problems."

Reply:
I get it now, thanks